xsnumpy/_numeric.py

```diff
@@ -0,0 +1,231 @@
+
+    The `ones` function creates an ndarray with the specified shape,
+    data type, and memory layout, initializing all its elements to zero.
+    This function is particularly useful for scenarios requiring a blank
+    array with known dimensions and type, where all elements must
+    initially be zero.
+
+    :param shape: The desired shape of the array. Can be an int for
+        1D arrays or a sequence of ints for multidimensional arrays.
+    :param dtype: The desired data type of the array, defaults to
+        `None` if not specified.
+    :param order: The memory layout of the array, defaults to `None`.
+    :return: An array initialized with ones with the specified
+        properties.
+    """
+    arr = empty(shape, dtype, order)
+    arr.fill(1)
+    return arr
+
+
+def ones_like(
+    a: ndarray,
+    dtype: DTypeLike | None = None,
+    order: None | _OrderKACF = None,
+) -> ndarray:
+    """Create a new array with the same shape and type as a given array,
+    filled with ones.
+
+    The `ones_like` function creates an array with the same shape as
+    the input array `a`. By default, the new array will have the same
+    data type as `a`, but this can be overridden with the `dtype`
+    parameter.
+
+    :param a: The reference array whose shape and optionally type are
+        used to create the new array.
+    :param dtype: The desired data type of the new array. If `None`,
+        the data type of `a` is used, defaults to `None`.
+    :param order: The desired memory layout for the new array, defaults
+        to `None`.
+    :return: A new array filled with ones, matching the shape of `a` and
+        the specified or inherited data type and memory layout.
+    """
+    dtype = a.dtype if dtype is None else dtype
+    return ones(a.shape, dtype, order)
+
+
+def eye(
+    size: int,
+    dtype: DTypeLike | None = None,
+    order: None | _OrderKACF = None,
+) -> ndarray:
+    """Create a 2-D identity matrix with ones on the main diagonal and
+    zeros elsewhere.
+
+    The `eye` function generates a square array of the specified size
+    with ones on its main diagonal (from the top-left to the bottom-
+    right corner) and zeros elsewhere. The data type and memory layout
+    of the array can be customized.
+
+    :param size: The number of rows and columns of the identity matrix.
+        This determines the shape `(size, size)` of the returned array.
+    :param dtype: The desired data type of the output array, defaults to
+        `None`.
+    :param order: The desired memory layout for the output array,
+        defaults to `None`.
+    :return: A square identity matrix of shape `(size, size)`.
+    """
+    if size <= 0:
+        raise ValueError("Size must be a positive integer")
+    arr = zeros((size, size), dtype, order)
+    for idx in range(size):
+        arr[idx, idx] = 1
+    return arr
+
+
+# TODO(xames3): Try to implement or support floating point step size.
```

Comment:
- this needs to be addressed immediately

Reply:
this is resolved in this commit here: https://github.com/xames3/xsnumpy/pull/7/commits/bf1794f99db43573250c5d70a8baea4f6ceb71eb